Code/Framework/AzCore/AzCore/DOM/DomBackend.cpp

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/DOM/DomBackend.h>
+
+namespace AZ::Dom
+{
+    Visitor::Result Backend::ReadFromBufferInPlace(char* buffer, Visitor& visitor)
+    {
+        return ReadFromBuffer(buffer, strlen(buffer), AZ::Dom::Lifetime::Persistent, visitor);
+    }
```

Comment:
:thumbsup: I went with an optional size parameter since you don't necessarily have to calculate it yourself if you don't have it (our JSON implementation won't care at all, for example)